sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -740,13 +740,22 @@ class MapboxMap :
    *
    * Map must be fully loaded for getting an altitude-compliant result if using 3D terrain.
    *
+   * If the screen coordinate is outside of the bounds of [MapView] the returned screen coordinate
+   * contains -1 for both coordinates.
+   *
    * @param coordinate A geographical coordinate on the map to convert to a screen coordinate.
    *
-   * @return Returns a screen coordinate on the screen in [MapOptions.size] platform pixels.
+   * @return Returns a screen coordinate on the screen in [MapOptions.size] platform pixels. If the screen coordinate is outside of the bounds of [MapView] the returned screen coordinate contains -1 for both coordinates.
    */
   override fun pixelForCoordinate(coordinate: Point): ScreenCoordinate {
     checkNativeMap("pixelForCoordinate")
-    return nativeMap.pixelForCoordinate(coordinate)
+    val coordinate = nativeMap.pixelForCoordinate(coordinate)
+    val screenSize = nativeMap.size
+    return if (coordinate.x in 0.0..screenSize.width.toDouble() && coordinate.y in 0.0..screenSize.height.toDouble()) {
```

Comment:
btw, do you know, creating range here will be converted to `0.0 <= coordinate.x <= screenSize.width.toDouble()` or not?
maybe it can be not super effective?

Reply:
we have parameterized tests to cover the corner cases here.

Reply:
tests is good, but here I'm talking about performance

Reply:
It's also suggested in https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/range-to.html.

According to the [source](https://github.com/JetBrains/kotlin/blob/6a670dc5f38fc73eb01d754d8f7c158ae0176ceb/libraries/stdlib/src/kotlin/ranges/Ranges.kt#L64-L97)

```
override fun contains(value: Double): Boolean = value >= _start && value <= _endInclusive
```

So I think it should be efficient to use.